tensorwaves/physics/helicityformalism/amplitude.py

```diff
@@ -548,7 +548,7 @@ def __init__(self, name: str, helicity: int,) -> None:
     @staticmethod
     def from_dict(definition: Dict[str, Any]) -> "_HelicityParticle":
         name = str(definition["Name"])
-        helicity = int(definition["Helicity"])
+        helicity = definition["Helicity"]
```

Comment:
Does this mean there's no type hint for the arguments of `_HelicityParticle`?

Reply:
Not sure what you mean by this. But I think mypy can correctly deduce the return type of `from_dict`

Reply:
The interface of `_HelicityParticle` looks like
```python
class _HelicityParticle:
    def __init__(self, name: str, helicity: int,) -> None:
        self.name: str = name
        self.helicity: int = helicity
```
So you see, it expect an `int` for `helicity`, while `helicity = definition["Helicity"]` would be `Any`.

Reply:
Oh dear, you are right! Good catch, I totally missed that...

Reply:
But what I don't understand is why `mypy` didn't fish this out...

Reply:
Because definition is of form `Dict[str, Any]` so Any -> int is no problem. Or am I missing something?

Reply:
Or should mypy mark `Any` -> `int` is a probem?

Reply:
Ah yes that's right... So I guess the lesson is to use `Any` sparingly

Reply:
And well... these `dict`s shouldn't be there anyway, but that's the recipe structure discussion again ;)

Reply:
Yeah, `Any` prevents the checking https://mypy.readthedocs.io/en/stable/kinds_of_types.html#the-any-type